### PR TITLE
[WEB-438] -Autocomplete improvement

### PR DESCRIPTION
**Pull request by NosiphoManyase**
### What problem is this solving?
This change is meant to encourage users to enter addresses using google autocomplete as well as . 

Desired change is not achieved exactly as required
Reason: You cannot access/manipulate returned google autocomplete addresses data until the users click on a suggested address (i.e through debouncing).

What has been achieved?

- **requests** for autocomplete addresses only executes **after 2 input value chars** are inserted by the user - this reduces requests
- if user backspaces to **less than 3 input chars**, google **autocomplete** and it's listener is **removed** to reduce requests as it stays mounted otherwise.
- To reduce premature error messaging for addresses  :
           1. A function **isSlowerNetwork** checks user network speed, if user is on slower network - **delay to error messages** is increased to 3s - from testing with throttled network this autocomplete response will not always be recieved in 3seconds.
           2. Before **checkAddressResults** function is **called** to evaluate if response has been received in order to display not found address message - I've put in logic to prevent checkAddressResults from executing if user is **still typing** **number**(house number) **or** has just put in a **space** in order to start typing street name. This will decrease premature address error message

#### How it works
Detailed above

#### How to test it?
- Test a few a few addresses and use checklist of what's been achieved above against current behaviour in prod.
- Test with fast and slow network.
- Check if code changes are efficient.

[Workspace](https://autocompleteweb436--thefoschini.myvtex.com/checkout/#/shipping)

### Screenshots/Video example usage:
- Desktop
- Tablet
- Movil
### Faster speed
https://github.com/user-attachments/assets/9cc648ba-d5ba-4ee1-a545-5f86b2fee2eb

### Throttled netwok
https://github.com/user-attachments/assets/62a9f9ee-78e7-4dcc-a0a7-0ba6681919c7

### CODE REVIEW

**Review by simmbiote — CHANGES_REQUESTED** [review]
@NosiphoManyase nice use of connection api!
I'm seeing a blank screen on the workspace at the moment. 
image

**Comment by NosiphoManyase** [comment]
> @NosiphoManyase nice use of connection api! I'm seeing a blank screen on the workspace at the moment. image

@simmbiote, I think it's because I put a link directly to checkout it returns blank if you're not logged in yet. Go to landing page and just let me know if you win

File: checkout-ui-custom/src/partials/Deliver/AddressForm.js [hunk 1]
```diff
@@ -186,12 +186,26 @@ const AddressForm = () => {
   <form id="bash--address-form" method="post">
     ${formFields}
 
+    <a
```
Comment (SegopotsoMakhutja) [inline checkout-ui-custom/src/partials/Deliver/AddressForm.js:189]: please pull the latest develop, these changes should not be there. i reverted this. @NosiphoManyase

File: checkout-ui-custom/src/utils/translations.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export default function setTranslations() {
-  $('.btn-go-to-shippping-method').text('Save address');
+  $('.btn-go-to-shippping-method').text('Delete address');
```
Comment (SegopotsoMakhutja) [inline checkout-ui-custom/src/utils/translations.js:2]: this should also not be here.

Comment (NosiphoManyase) [reply checkout-ui-custom/src/utils/translations.js:2]: Thanks, let me pull latest changes

**Review by SegopotsoMakhutja — APPROVED** [review]

**Pull request closed without merging**